### PR TITLE
feat: add encrypt values for secret section

### PR DESCRIPTION
**Pull request by akastav**
Problem:
we prepare passwords via terraform and use bcrypt to encrypt the password (example of the encrypted password is "$2a$10$H1a30nMr9v2QE2nkyz0BoOD2J0I6FQFMtHS0csEg12RBWzfRuuoE6". When ArgoCD creates a secret and tries to encrypt in base64, characters like $ are interpreted by the shell as variables and breaks the password.

Solutions:
This will help to pass the prepared secrets encrypted via bcrypt to the chart parameters. If ".Values.configs.secret.encrypt" = false, used stringData without base64 encrypt procedure.

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [X] I have updated the chart version in `Chart.yaml` following Semantic Versioning.
* [X] Any new values are backwards compatible and/or have sensible default.
* [X] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [X] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [ ] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

File: charts/argo-cd/templates/argocd-configs/argocd-secret.yaml [hunk 1]
```diff
@@ -14,7 +14,11 @@ metadata:
 type: Opaque
 {{- if or .Values.configs.secret.githubSecret (or .Values.configs.secret.gitlabSecret .Values.configs.secret.bitbucketUUID .Values.configs.secret.bitbucketServerSecret .Values.configs.secret.gogsSecret .Values.configs.secret.argocdServerAdminPassword .Values.configs.secret.argocdServerTlsConfig .Values.configs.secret.extra) }}
 # Setting a blank data again will wipe admin password/key/cert
+{{- if .Values.configs.secret.encrypt }}
 data:
+{{- else }}
+stringData:
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/argocd-configs/argocd-secret.yaml:20]: This will not work, look at some lines below. We use `| b64enc` everywhere.

**Comment by mkilchhofer** [comment]
Hi @akastav 
How do you pass that TF encrypted bcrypt value? Do you use the helm provider? We also use TF to deploy this chart and do not have this issue:
```terraform
data "template_file" "oidc_config_template" {
  template = file("${path.module}/helm_values.yaml")
  vars = {
    // (..)
    argocd_password              = bcrypt(var.argocd_password)
    // (..)
  }
}

resource "helm_release" "argocd" {
  name       = var.argocd_helm_release_name
  repository = var.argocd_helm_repository
  chart      = var.argocd_helm_chart_name
  version    = var.argocd_helm_chart_version
  namespace  = var.argocd_namespace

  values = [data.template_file.oidc_config_template.rendered]

  depends_on = [kubernetes_namespace.argocd]

}
```

**Comment by akastav** [comment]
Hi @mkilchhofer! 
We don't use helm provider, in our TF project Swiss-army-kube we create random pass https://github.com/provectus/sak-argocd/blob/master/main.tf#L129 and store it in system manager, param store.  For decrypt pass, we used https://github.com/provectus/sak-argocd/blob/master/main.tf#L262 python code as configManagementPlugins. After all, we create ArgoCD spec and push it to git.  Some kind of self-managed ArgoCD

File: charts/argo-cd/templates/argocd-configs/argocd-secret.yaml (hunk 1 above)
Comment (akastav) [reply charts/argo-cd/templates/argocd-configs/argocd-secret.yaml:20]: Maybe use something like
`admin.password: {{- if .Values.configs.secret.encrypt }}{{ .Values.configs.secret.argocdServerAdminPassword | b64enc }} {{else}} {{ .Values.configs.secret.argocdServerAdminPassword }}`

**Comment by mkilchhofer** [comment]
> Hi @mkilchhofer!
> We don't use helm provider, in our TF project Swiss-army-kube we create random pass https://github.com/provectus/sak-argocd/blob/master/main.tf#L129 and store it in system manager, param store. For decrypt pass, we used https://github.com/provectus/sak-argocd/blob/master/main.tf#L262 python code as configManagementPlugins. After all, we create ArgoCD spec and push it to git. Some kind of self-managed ArgoCD

Oh I see. This have to do with `values` vs. `parameters`:
```yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: argocd-testing
spec:
  destination:
    namespace: testing
    server: https://kubernetes.default.svc
  project: default
  source:
    chart: argo-cd
    helm:
      parameters:
      - name: configs.secret.argocdServerAdminPasswordDD
        value: $2a$10$H1a30nMr9v2QE2nkyz0BoOD2J0I6FQFMtHS0csEg12RBWzfRuuoE6
      values: |
        configs:
          secret:
            argocdServerAdminPassword: $2a$10$H1a30nMr9v2QE2nkyz0BoOD2J0I6FQFMtHS0csEg12RBWzfRuuoE6
```
It works in `values` but not in `parameters` as parameters supports variable substitution.

But I think your solution would not help. This is a feature of ArgoCD which unfortunately introduces this.
What about escaping the dollar sign with two or using the `values` attribute?

**Comment by mkilchhofer** [comment]
Ah! This indeed would help. Base64 cannot contain a `$`

**Comment by akastav** [comment]
Thanks for the hint about values - I'll definitely try it

**Comment by mkilchhofer** [comment]
> Thanks for the hint about values - I'll definitely try it
> 
> @mkilchhofer Thx it works for us! This way much easy and will not allow you to compromise other accidentally unencrypted passwords. You can close this PR, but add this hint to the FAQ ;)

👍 I created an issue so we do not forget this: https://github.com/argoproj/argo-helm/issues/785

**Pull request closed without merging**